storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerConfig.java

```diff
@@ -188,10 +186,14 @@ public final class RemoteLogManagerConfig {
     public static final String REMOTE_FETCH_MAX_WAIT_MS_DOC = "The maximum amount of time the server will wait before answering the remote fetch request";
     public static final int DEFAULT_REMOTE_FETCH_MAX_WAIT_MS = 500;
 
-    public static final ConfigDef CONFIG_DEF = new ConfigDef();
+    public static final ConfigDef CONFIG_DEF;
+
+    public static ConfigDef configDef() {
+        return CONFIG_DEF;
```

Comment:
This is still a mutable shared object. configDef() should execute all of the code in the static initializer each time it is called to produce a new ConfigDef.

Reply:
@muralibasani Please see https://github.com/apache/kafka/blob/8b3c77c6711d8c1424659a1b4621a8c823fa2236/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java#L201

CONFIG_DEF is still mutable and shouldn't exist.

Reply:
Got it, updated it similar to ConnectorConfig

Reply:
However in AdminClientConfig and a few other configs, I notice config objects are mutable

Reply:
It's a pretty common mistake it seems. It is more of a risk when working with arbitrary plugins, but is still good practice.

It would have been nice to have ConfigDef be immutable, and have the define() methods on a ConfigDefBuilder instead. But the design we have now is very entrenched, so we can just cope with it.